hibernate-reactive-core/src/main/java/org/hibernate/reactive/session/impl/ReactiveCriteriaQueryImpl.java

```diff
@@ -0,0 +1,145 @@
+package org.hibernate.reactive.session.impl;
+
+import org.hibernate.AssertionFailure;
+import org.hibernate.jpa.spi.HibernateEntityManagerImplementor.QueryOptions;
+import org.hibernate.query.criteria.internal.CriteriaBuilderImpl;
+import org.hibernate.query.criteria.internal.CriteriaQueryImpl;
+import org.hibernate.query.criteria.internal.QueryStructure;
+import org.hibernate.query.criteria.internal.SelectionImplementor;
+import org.hibernate.query.criteria.internal.ValueHandlerFactory;
+import org.hibernate.query.criteria.internal.compile.ImplicitParameterBinding;
+import org.hibernate.query.criteria.internal.compile.InterpretedParameterMetadata;
+import org.hibernate.query.criteria.internal.compile.RenderingContext;
+import org.hibernate.reactive.session.ReactiveQuery;
+import org.hibernate.reactive.session.ReactiveSession;
+import org.hibernate.type.Type;
+
+import javax.persistence.TypedQuery;
+import javax.persistence.criteria.Selection;
+import java.lang.reflect.Field;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * A reactific {@link CriteriaQueryImpl}, providing the implementation
+ * of {@link javax.persistence.criteria.CriteriaQuery}.
+ *
+ * @author Gavin King
+ */
+public class ReactiveCriteriaQueryImpl<T> extends CriteriaQueryImpl<T> {
+
+	//TODO: expose this field in ORM!
+	private static Field queryStructureField;
+	static {
+		try {
+			queryStructureField = CriteriaQueryImpl.class.getDeclaredField("queryStructure");
+			queryStructureField.setAccessible(true);
+		}
+		catch (Exception e) {
+			throw new AssertionFailure("missing field", e);
+		}
+	}
```

Comment:
We can't use reflection in Quarkus so we definitely need to expose this field in ORM.

Reply:
See https://github.com/hibernate/hibernate-orm/pull/3419